tests/testcase.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
```

Comment:
This is going to outgrow the (my) accepted level of shell script usage very quickly.

Would you object to me re-writing this in python at some point?

Reply:
No objections. I find what these scripts are doing simpler to write in bash than python, but I'm ok with them being written in python.